core/src/main/java/org/mapfish/print/servlet/job/impl/hibernate/HibernateJobQueue.java

```diff
@@ -209,11 +214,26 @@ public final void shutdown() {
 
   private void cleanup() {
     TransactionTemplate tmpl = new TransactionTemplate(this.txManager);
-    final int nbDeleted =
-        tmpl.execute(
-            status ->
-                HibernateJobQueue.this.dao.deleteOld(
-                    System.currentTimeMillis() - getTimeToKeepAfterAccessInMillis()));
-    metricRegistry.counter(getClass().getName() + ".deleted_old").inc(nbDeleted);
+    String name = MetricRegistry.name(getClass().getSimpleName(), "deletedOldPrintJobs");
+    try (Timer.Context deletion = metricRegistry.timer(name).time()) {
+      final Integer nbDeleted =
+          tmpl.execute(
+              status -> {
+                try {
+                  return HibernateJobQueue.this.dao.deleteOld(
+                      System.currentTimeMillis() - getTimeToKeepAfterAccessInMillis());
+                } catch (Exception e) {
+                  unhealthyCountersHealthCheck.recordUnhealthyProblem(
+                      name, "failedToDeleteOldJobs");
+                  return null;
+                }
+              });
+      if (deletion.stop() > CLEAN_UP_INTERVAL_TIME_UNIT.toNanos(cleanupInterval)) {
+        unhealthyCountersHealthCheck.recordUnhealthyProblem(name, "tooManyJobsToDeleteInInterval");
+      }
```

Comment:
I'm not sure about that...

Reply:
The purpose is to know the server is constantly running out of time in the deletion process. So plenty of needless scheduling events are recorded in the cleanUpTimer threadpool, since the scheduler will only run one at a time.

Reply:
Looks good, why not also add a warning message in the logs?

Reply:
Guillaume decided it was an acceptable behavior, and didn't log it despite the fact that all the other RuntimeExceptions below were logged. That's the only reason I did not log it. But my strategy is to record in a counter the number of occurrences of this ignored issue to see if they happen far too much to be ignored. In that case, we might need to put a threshold when the rate of occurrence gets too high.
Either way I am fine.